patches/minecraft/net/minecraft/resources/ResourceLocation.java.patch

```diff
@@ -1,5 +1,23 @@
 --- a/net/minecraft/resources/ResourceLocation.java
 +++ b/net/minecraft/resources/ResourceLocation.java
+@@ -34,6 +_,8 @@
+       this.path = p_249394_;
+    }
+ 
++   /** @deprecated Forge: Use {@link #fromNamespaceAndPath(String, String)} instead */
++   @Deprecated(forRemoval = true, since = "1.21")
+    public ResourceLocation(String p_135811_, String p_135812_) {
```

Comment:
I'd skip them deprecation, as its a mojang thing. But pointing them to the new one in the comments isnt bad.